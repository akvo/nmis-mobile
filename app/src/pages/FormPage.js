import React from 'react';
import { FormContainer } from '../form';
import { BaseLayout } from '../components';
import { FormState } from '../store';

// TODO:: todelete
import * as formDefinition from '../form/example-form.json';

const FormPage = ({ navigation, route }) => {
  const selectedForm = FormState.useState((s) => s.selectedForm);

  const formJSON = React.useMemo(() => {
    if (!selectedForm?.json) {
      return formDefinition;
    }
    return JSON.parse(selectedForm.json.replace(/''/g, "'"));
  }, [selectedForm]);

  const goBack = () => {
    navigation.navigate('ManageForm', { ...route?.params });
  };

  return (
    <BaseLayout title={route?.params?.name} back={goBack}>
      <FormContainer forms={formJSON || formDefinition} initialValues={{}} />
    </BaseLayout>
  );
};

export default FormPage;
