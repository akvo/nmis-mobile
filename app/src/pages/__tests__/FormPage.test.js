import React from 'react';
import renderer from 'react-test-renderer';
jest.useFakeTimers();
import FormPage from '../FormPage';

describe('FormPage component', () => {
  test('should render component correctly', () => {
    const tree = renderer.create(<FormPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
