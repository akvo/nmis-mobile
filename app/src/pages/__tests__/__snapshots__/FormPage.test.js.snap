// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormPage component should render component correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#f9fafb",
      "flex": 1,
      "flexDirection": "column",
      "flexWrap": "wrap",
    }
  }
  testID="stack-container"
>
  <mock-FormContainer>
    <button
      onPress={[Function]}
      testID="mock-submit-button"
    >
      Submit
    </button>
  </mock-FormContainer>
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    style={
      {
        "width": "100%",
      }
    }
    testID="form-page-dialog-menu"
    transparent={true}
    visible={false}
  />
</View>
`;
